src/core/FlickingError.ts

```diff
@@ -30,7 +32,8 @@ class FlickingError extends Error {
     public message: string,
     public code: number) {
     super(message);
-    Object.setPrototypeOf(this, FlickingError.prototype);
+
+    setPrototypeOf(this, FlickingError.prototype);
```

Comment:
By doing setPrototypeOf, what functionality do you expect?

Reply:
While making this, there were a few restrictions. I don't remember everything but here's some
- a thrown `err instanceof FlickingError` should be true
- a thrown `err instanceof Error` also should be true
- FlickingError should have `message`, `stack` and `code` property
- FlickingError should be throwable & catchable

Reply:
Can it be seen that this cannot be solved by simply inheriting the Error class?

Reply:
Yes, that doesn't work for some reason...
I remember it couldn't solve one of the restrictions above, I guess the first one was?

Reply:
I don't know why. Is it because it is not a native class because it is transpiled?

Reply:
If I remove that, `expect(err).to.be.instanceOf(FlickingError);` fails.

As you said, it's not a native class after it's transpiled.

> BEF remove
```js
    var FlickingError = function (_super) {
      __extends$1(FlickingError, _super);
      /**
       * @param message Error message<ko>에러 메시지</ko>
       * @param code Error code<ko>에러 코드</ko>
       */


      function FlickingError(message, code) {
        var _this = _super.call(this, message) || this;

        _this.message = message;
        _this.code = code;
        setPrototypeOf(_this, FlickingError.prototype);
        _this.name = "FlickingError";
        return _this;
      }

      return FlickingError;
    }(Error);
```

> AFT remove
```js
    var FlickingError = function (_super) {
      __extends$1(FlickingError, _super);
      /**
       * @param message Error message<ko>에러 메시지</ko>
       * @param code Error code<ko>에러 코드</ko>
       */


      function FlickingError(message, code) {
        var _this = _super.call(this, message) || this;

        _this.message = message;
        _this.code = code;
        _this.name = "FlickingError";
        return _this;
      }

      return FlickingError;
    }(Error);
```


```js
    function __extends$1(d, b) {
      if (typeof b !== "function" && b !== null) throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
      extendStatics$1(d, b);

      function __() {
        this.constructor = d;
      }

      d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
```

By that, `expect(err).to.be.instanceOf(FlickingError);` fails.